src/components/Wizard/CreateVmWizard/providers/v2vvmwareController.js

```diff
@@ -0,0 +1,171 @@
+import { DeploymentModel, RoleModel, ServiceAccountModel, RoleBindingModel } from '../../../../models';
+import { V2VVMWARE_DEPLOYMENT_NAME } from '../constants';
+
+// eslint-disable-next-line no-console
+const rejectLogger = title => reason => console.warn(`Failed to create ${title}, reason`, reason);
+
+// The controller is namespace-scoped, especially due to security reasons
+// Let's make sure its started within the desired namespace (which is not by default).
+// The V2VVmware CRD is expected to be already created within the cluster (by Web UI installation)
+export const startV2VVMWareController = async ({ k8sCreate, k8sGet, namespace }) => {
+  try {
+    await k8sGet(DeploymentModel, V2VVMWARE_DEPLOYMENT_NAME, namespace);
+  } catch {
+    // Deployment does not exist or the does not have permissions to see Deployments in this namespace
+    // TODO: notify the user in other way not just the console.log
+
+    // eslint-disable-next-line no-console
+    console.info('V2V VMWare controller deployment not found, so creating one ...');
+    const params = { k8sCreate, namespace };
+
+    /* eslint-disable promise/catch-or-return */
+    createServiceAccount(params).then(undefined, rejectLogger('Service Account'));
+    createRole(params).then(undefined, rejectLogger('Role'));
+    createRoleBinding(params).then(undefined, rejectLogger('Role Binding'));
+    createOperator(params).then(undefined, rejectLogger('V2V VMWare controller'));
+    /* eslint-enable promise/catch-or-return */
+  }
+};
+
+const createServiceAccount = ({ k8sCreate, namespace }) =>
+  k8sCreate(ServiceAccountModel, {
+    apiVersion: ServiceAccountModel.apiVersion,
+    kind: ServiceAccountModel.kind,
+    metadata: {
+      name: V2VVMWARE_DEPLOYMENT_NAME,
+      namespace,
+    },
+  });
+
+const createRole = ({ k8sCreate, namespace }) =>
+  k8sCreate(RoleModel, {
```

Comment:
I understand your point, it would be worth doing that in a more generic case.
But all these objects are for this very specific usecase only. And this `v2vvmwareController.js` handles that case only.

Honestly, I do not like having this deployment logic in UI.
But I did not find any other place where to put it.

From architecture perspective, good place can be:
- web-ui operator (but privileges issues)
- specific "namespace-preparation" code (does not exist ATM, would prepare a namespace for virtualization)

I've intentionally created the controller to be namespaced (not cluster-wide), mostly due to security reasons.

Reply:
Ok,I guess, if the meaning is to have them encapsulated in one file.